blockchain/My-Wallet-V3-iOS
# feat(Homebrew-Shapeshift): update values from conversion socket message

Please merge #482 before this one.

## Objective

Update labels from conversion socket messages.

## Description

### User Input

 
- `PrimaryLabel` is the larger label - this shows the current input (what the user is typing). This label's text is assigned to `ExchangeInputs`'s `activeInput.input`.
- `SecondaryLabel` is the smaller label below the `PrimaryLabel` - this also shows the current input, but in crypto or fiat (whatever is opposite to the kind `PrimaryLabel` is displaying) - this label's text is assigned to `ExchangeInputs`'s `lastOutput`.
- The `ExchangeConversionService` is a class that helps with sanitizing data from a `Conversion` struct.
- The `Conversion` socket message is received to the `PublishSubject` and is sanitized by removing the trailing zeros and possibly trailing decimal point (performed by the `ExchangeConversionService`).
- The `SecondaryLabel`'s text is updated with the `output` value.
- The sanitized `input` value is compared to a sanitized copy of the current `activeInput.input` - if the values are the same, then the label is not updated to allow uninterrupted keypad input (so that the user can keep deleting zeros from `50.00`, for example, but the output value is still the latest from the socket).
- This still needs some tweaks, since in some cases user input is still being interrupted.

### TradingPairView 
- `TradingPairView` is (somewhat) hooked up through changing the button titles to reflect crypto values (also provided by the `ExchangeConversionService`).



Especially interested in feedback on the `ExchangeConversionService`.

## How to Test

- Revert 6e49088e24b0012d841c12a39870bdca0fbc4b2b
- Open Exchange
- Start typing in values on the keypad and see that the `SecondaryLabel` updates. Can also tap the right-hand side button to switch between crypto and fiat.

## Screenshot/Design assessment
Measurements are still off, but getting closer to design specs. GIF is sped up for easier viewing.
![56yqkT](https://i.makeagif.com/media/9-11-2018/56yqkT.gif)

## Merge Checklist
